runtime/util/vmargs.c

```diff
@@ -76,8 +76,46 @@
 
 #define LARGE_STRING_BUF_SIZE 256
 
+struct JavaLauncherOption {
+	char *optionName;
+	int optionNameLength;
+};
+
+/* options in this table will be not treated as a user Argument 
+Note, these options need to be in lexical order (ASCII values) for the binary search function to properly work */
+static const struct JavaLauncherOption JavaLauncherOptionTable[] = {
+	{"-Djava.class.path=", strlen("-Djava.class.path=")},
+	{"-Dsun.java.command=", strlen("-Dsun.java.command=")},
+	{"-Dsun.java.launcher.pid=", strlen("-Dsun.java.launcher.pid=")},
+	{"-Dsun.java.launcher=", strlen("-Dsun.java.launcher=")},/* Note, 'launcher=' is placed after .launcher.pid` since '=' has a higher ASCII value than '.' */
+	{"-Xjcl:", strlen("-Xjcl:")},
+};
+
+#define ignoredOptionTableEnd (sizeof(JavaLauncherOptionTable) / sizeof(struct JavaLauncherOption) - 1)
+
 static char * getStartOfOptionValue(J9VMInitArgs* j9vm_args, IDATA element, const char *optionName);
 
+UDATA
+isGeneratedArgument(const char *optionName) {
+	IDATA start = 0;
+	IDATA end = ignoredOptionTableEnd;
+	IDATA mid = 0;
+	while (start <= end) {
+		mid = (start + end) / 2;
+		IDATA optionCompare = strncmp(optionName, JavaLauncherOptionTable[mid].optionName, JavaLauncherOptionTable[mid].optionNameLength);
```

Comment:
This probably won't compile on Windows.  declare `optionCompare` before any executable code.  `compareResult` would be amore meaningful name.